### ISSUE TITLE
partial_sort_copy is underconstrained

### ISSUE THREAD

**Issue by CaseyCarter** [ericniebler/stl2#128]
The declaration of `partial_sort_copy` in the TS:
```c++
template<InputIterator I1, Sentinel<I1> S1, RandomAccessIterator I2, Sentinel<I2> S2,
    class Comp = less<>, class Proj = identity>
  requires IndirectlyCopyable<I1, I2>() && Sortable<I2, Comp, Proj>()
I2 partial_sort_copy(I1 first, S1 last, I2 result_first, S2 result_last,
                     Comp comp = Comp{}, Proj proj = Proj{});
```
accepts two ranges. In addition to comparing two elements of the second range - which is properly constrained - implementations need to compare elements of the first range with elements of the second. To provide that capability we need an additional requirement along the lines of `IndirectCallableStrictWeakOrder<Comp, I1, Projected<I2, Proj>>()` or, if the intent is that the single projection be used for both ranges, `IndirectCallableStrictWeakOrder<Comp, Projected<I1, Proj>, Projected<I2, Proj>>()`. A second projection would be necessary to provide full generality:
```c++
template<InputIterator I1, Sentinel<I1> S1, RandomAccessIterator I2, Sentinel<I2> S2,
    class Proj1 = identity, class Proj2 = identity,
    IndirectCallableStrictWeakOrder<Projected<I1, Proj1>, Projected<I2, Proj2>> Comp = less<>>
  requires IndirectlyCopyable<I1, I2>() && Sortable<I2, Comp, Proj2>()
I2 partial_sort_copy(I1 first, S1 last, I2 result_first, S2 result_last,
                     Comp comp = Comp{}, Proj1 proj1 = Proj1{}, Proj2 proj2 = Proj2{});
```
which happens to be roughly equivalent to the declaration in range-v3.

**Comment by CaseyCarter** [comment]
...or is the effect of `IndirectlyCopyable`'s requirement:

> If `Out` is `Readable`, after the assignment `*o = *i`, `*o` is equal to the value of `*i`.

to constrain the ranges being copied to have the same `ValueType`, so that `Sortable`'s requirement:

> `IndirectCallableStrictWeakOrder<R, Projected<I, P>>()`

implies `IndirectCallableStrictWeakOrder<Comp, Projected<I1, Proj>, Projected<I2, Proj>>()` in `partial_sort_copy`?

**Comment by CaseyCarter** [comment]
I suppose real the question here is whether `IndirectlyCopyable<I1, I2>` should forbid conversions as is currently specified, or if that requirement should be changed to "After the assignment `*o = *i`, if `Out` is `Readable` then `*o == *i`".

**Comment by ericniebler** [comment]
> I suppose real the question here is whether `IndirectlyCopyable<In, Out>` should forbid conversions as is currently somewhat obliquely specified [...]

I don't see that in the specification. I see the use of the "is equal to" Phrase of Power. That is appealing to an equality that is out of the system -- the same Phrase of Power that is used to define `EqualityComparable`, but without requiring it.

**Comment by ericniebler** [comment]
Fixed in ba418b3
